### PR TITLE
[Hub Generated] Review request for Microsoft.Insights to add version stable/2021-05-01

### PR DESCRIPTION
**Pull request by ToddKingMSFT**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/ToddKingMSFT/azure-rest-api-specs/dev-monitor-Microsoft.Insights-2021-05-01...master/).

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [x] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [x] SDK of .NET (need service team to ensure code readiness)
      - [x] SDK of Python
      - [x] SDK of Java
      - [x] SDK of Js
      - [x] SDK of Go
      - [x] PowerShell
      - [x] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [ ] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by ToddKingMSFT** [comment]
@kairu-ms / @weidongxu-microsoft Can one of you help me complete this pull request?

**Comment by ToddKingMSFT** [comment]
@kairu-ms / @weidongxu-microsoft Can one of you review this pull request?

**Comment by weidongxu-microsoft** [comment]
@ToddKingMSFT If you have new API, please mark it on the checkbox, so that ARM can be invovled.

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2021-05-01/metrics_API.json [hunk 1]
```diff
@@ -0,0 +1,803 @@
...
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Insights/metrics": {
+      "get": {
+        "tags": [
+          "Metrics"
+        ],
+        "operationId": "MultiResourceMetrics_List",
```
Comment (weidongxu-microsoft) [inline specification/monitor/resource-manager/Microsoft.Insights/stable/2021-05-01/metrics_API.json:41]: A seemingly more common naming pattern would be `Metrics_ListAtSubscriptionScope`

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2021-05-01/metrics_API.json [hunk 2]
```diff
@@ -0,0 +1,803 @@
...
+          },
+          "Get Metric with error": {
+            "$ref": "./examples/GetMetricError.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "BodyParameters": {
```
Comment (weidongxu-microsoft) [inline specification/monitor/resource-manager/Microsoft.Insights/stable/2021-05-01/metrics_API.json:275]: Maybe a more specific name (if List really uses a body). `BodyParameters` is not meaningful when this JSON is used together with 10+ other JSONs.

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2021-05-01/metricDefinitions_API.json [hunk 3]
```diff
@@ -0,0 +1,354 @@
...
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Insights/metricDefinitions": {
+      "get": {
+        "tags": [
+          "MetricDefinitions"
+        ],
+        "operationId": "MultiResourceMetricDefinitions_List",
```
Comment (weidongxu-microsoft) [inline specification/monitor/resource-manager/Microsoft.Insights/stable/2021-05-01/metricDefinitions_API.json:41]: Similarly, maybe `MetricDefinitions_ListAtSubscriptionScope`

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2021-05-01/metrics_API.json (hunk 1 above)
Comment (ToddKingMSFT) [reply specification/monitor/resource-manager/Microsoft.Insights/stable/2021-05-01/metrics_API.json]: I've incorporated this feedback

**Comment by ToddKingMSFT** [comment]
@weidongxu-microsoft I guess it's kind of complicated.
This API was originally introduced as a preview API in December 2017. And it's been a preview API since then until recently S360 started firing on us telling us it has been in preview too long. So this PR is to release that same API as a stable GA version.
Because it was never previously documented in a swagger, it looks like a brand new API, but in reality we're just taking the existing API and documenting it as a newer GA version. I'm not sure I have much latitude to change anything with it as that would represent a breaking change to all the users already using it today.

**Comment by ToddKingMSFT** [comment]
A different PR with the same content was already checked in

**Pull request closed without merging**